### PR TITLE
Support java.util.Date in TestGen

### PR DESCRIPTION
**Pull request by rsynek**
This modification was necessary for me to proceed in isolation of NPE in Drools.

### CODE REVIEW

**Comment by ge0ffrey** [comment]
Would we want similar support for LocalDate, LocalDateTime and LocalTime? (or Temporal in general)
Doesn't need to be in this PR - just thinking aloud.
This PR is good to merge.

**Comment by rsynek** [comment]
@ge0ffrey `LocalDate` and others work because they have `parse` method and some types of fields are created by having that method. See https://github.com/kiegroup/optaplanner/blob/master/optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/fact/TestGenValueFact.java#L98

I would also prefer waiting for @yurloc before I finish this PR (there is no coverage for the new code).

**Comment by ge0ffrey** [comment]
Given that users don't use the testgen framework, I don't mind there being no coverage. It's always nice to have of course.

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/fact/TestGenDateValueProvider.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.director.drools.testgen.fact;
+
+import java.util.Date;
+
+public class TestGenDateValueProvider extends TestGenAbstractValueProvider<Date> {
```
Comment (yurloc) [inline optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/fact/TestGenDateValueProvider.java]: Should be package-private.

Comment (ge0ffrey) [reply optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/fact/TestGenDateValueProvider.java]: Done through github suggestion.

**Comment by yurloc** [comment]
For the record, this shouldn't have been merged. I suspect that the new `Date` provider class behaves incorrectly. We agreed with @rsynek that he would add a test that prove my concern and he would change the implementation accordingly.

Since TestGen is very low priority, let's address this later (when someone hits the bug in `TestGenDateValueProvider` :slightly_smiling_face:).

**Comment by ge0ffrey** [comment]
@yurloc My apologies for jumping the gun. I am trying to go through the PR list and getting the PR's merged (good is the enemy of perfect - sometimes). Further issues can be handled in incremental PR's.

**Comment by rsynek** [comment]
My bad - I should have updated this conversation. Nevermind, I will prepare a new PR with the remaining work.

**Comment by rsynek** [comment]
Finished in https://github.com/kiegroup/optaplanner/pull/523

**Pull request merged**